core/src/SYCL/Kokkos_SYCL_GraphNodeKernel.hpp

```diff
@@ -0,0 +1,174 @@
+          typename... Args>
+class GraphNodeKernelImpl<Kokkos::Experimental::SYCL, PolicyType, Functor,
+                          PatternTag, Args...>
+    : public PatternImplSpecializationFromTag<
+          PatternTag, Functor, PolicyType, Args...,
+          Kokkos::Experimental::SYCL>::type {
+ public:
+  using Policy       = PolicyType;
+  using graph_kernel = GraphNodeKernelImpl;
+  using base_t       = typename PatternImplSpecializationFromTag<
+      PatternTag, Functor, Policy, Args..., Kokkos::Experimental::SYCL>::type;
+
+  // TODO use the name and executionspace
+  template <typename PolicyDeduced, typename... ArgsDeduced>
+  GraphNodeKernelImpl(std::string, Kokkos::Experimental::SYCL const&,
+                      Functor arg_functor, PolicyDeduced&& arg_policy,
+                      ArgsDeduced&&... args)
+      : base_t(std::move(arg_functor), (PolicyDeduced &&) arg_policy,
+               (ArgsDeduced &&) args...) {}
+
+  template <typename PolicyDeduced>
+  GraphNodeKernelImpl(Kokkos::Experimental::SYCL const& exec_space,
+                      Functor arg_functor, PolicyDeduced&& arg_policy)
+      : GraphNodeKernelImpl("", exec_space, std::move(arg_functor),
+                            (PolicyDeduced &&) arg_policy) {}
+
+  ~GraphNodeKernelImpl() {
+    if (m_driver_storage) {
+      Kokkos::Experimental::SYCLDeviceUSMSpace().deallocate(m_driver_storage,
+                                                            sizeof(base_t));
+    }
+  }
+
+  void set_sycl_graph_ptr(
+      sycl::ext::oneapi::experimental::command_graph<
+          sycl::ext::oneapi::experimental::graph_state::modifiable>*
+          arg_graph) {
+    m_graph_ptr = arg_graph;
+  }
+
+  void set_sycl_graph_node_ptr(
+      std::optional<sycl::ext::oneapi::experimental::node>* arg_node) {
+    m_graph_node_ptr = arg_node;
+  }
+
+  std::optional<sycl::ext::oneapi::experimental::node>& get_sycl_graph_node()
+      const {
+    return *m_graph_node_ptr;
+  }
+
+  sycl::ext::oneapi::experimental::command_graph<
+      sycl::ext::oneapi::experimental::graph_state::modifiable>&
+  get_sycl_graph() const {
+    return *m_graph_ptr;
+  }
+
+  Kokkos::ObservingRawPtr<base_t> allocate_driver_memory_buffer() const {
+    KOKKOS_EXPECTS(m_driver_storage == nullptr);
+    m_driver_storage = static_cast<base_t*>(
+        Kokkos::Experimental::SYCLDeviceUSMSpace().allocate(
+            "GraphNodeKernel global memory functor storage", sizeof(base_t)));
+    KOKKOS_ENSURES(m_driver_storage != nullptr);
+    return m_driver_storage;
+  }
+
+ private:
+  Kokkos::ObservingRawPtr<sycl::ext::oneapi::experimental::command_graph<
+      sycl::ext::oneapi::experimental::graph_state::modifiable>>
+      m_graph_ptr = nullptr;
+  Kokkos::ObservingRawPtr<std::optional<sycl::ext::oneapi::experimental::node>>
+      m_graph_node_ptr                          = nullptr;
+  Kokkos::OwningRawPtr<base_t> m_driver_storage = nullptr;
```

Comment:
I don't follow what we need that storage for.
```suggestion
```

Reply:
Good catch! Removed.